api/src/main/scala/pl/touk/nussknacker/engine/api/process/RunMode.scala

```diff
@@ -3,16 +3,21 @@ package pl.touk.nussknacker.engine.api.process
 /**
   * Specifies the mode a node is invoked. It can be one of the following values:
   * <ul>
-  * <li>Normal - process is executed inside Flink or other engine. Real implementation of node should be used. Note, this
-  * mode is also used for service query, validations and test data generation.</li>
+  * <li>Normal - used when process is executed inside Flink or other engine, and also for service query, validations and test data generation.
+  * In this mode real implementation of node should be used.</li>
   * <li>Test - process is run in test mode to collect results for test data. Can be used to stub real implementation.</li>
   * </ul>
   */
 sealed trait RunMode
 
 object RunMode {
 
-  case object Normal extends RunMode
+  sealed trait Normal extends RunMode
```

Comment:
Do u see some advantage in grouping four modes below to "normal" mode other than for backward compatibility? IMO it is not straightforward why e.g. TestDataGeneration is normal mode and test is test mode

Reply:
No, it was only for sake of backwards compatibility.